### DISCUSSION TITLE
Report Error should first be reported at project level

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#17259] [Feature Requests]
<img width="545" alt="Screenshot 2023-01-21 at 10 53 07" src="https://user-images.githubusercontent.com/6641242/213861631-c6dc64da-7822-4d7b-9d2d-10a70a3a03a5.png">

Currently this "Report Error" button opens up an issue on the Directus repo.

Could it rather be configurable such that it opens an issue on the project's repo first, such that it could first be debugged there before involving and burdening the Directus Team with all such (possibly minor) issues?

**Comment by rijkvanzanten** [comment]
Good one! What would be a good way to configure the report error interaction? Maybe a field in project-settings? If we leave it as just a link, you could also do a `mailto:` address I suppose, which would be a nice way to support both email and other things 🤔

**Reply by paescuj** [reply]
Good question! I think it would indeed require a dedicated field to customize this link.
In order to be really useful, it would also require a mapping of error / context data to params, for example by allowing a templated link like `https://my-custom-report-link.com/issue?error={{error}}&version={{directusVersion}}` (would also work for `mailto:`).

**Comment by u12206050** [comment]
That sounds good. Either in env or in project settings.

Shall I make a PR or is it pretty easy for core team to do?

**Reply by rijkvanzanten** [reply]
Lets figure out how we want this last bit to work first:

> it would also require a mapping of error / context data to params, for example by allowing a templated link like https://my-custom-report-link.com/issue?error={{error}}&version={{directusVersion}} (would also work for mailto:).

What parameters do/should we have available? Should it match `/server/info`?

**Reply by u12206050** [reply]
I think the following would make sense.

- Current page path/url
- error
- user id (that got error)

Since the backend isn't different per project i don't see how version or server info could help. But any stack or context (eg. Browser info) could be useful, depending on the error.

**Reply by rijkvanzanten** [reply]
Is that an exhaustive list? Anything else we should spec out? I'm wondering if there's a more "standard" set that we could rely on so we don't have to document / come up with yet another object structure that's vaguely similar 🤔 

Maybe we could say:

- Context
  - Browser UA
  - Current Page
- Error
- User
  - (Anything from the user object / user store)
- Role
  - (Anything from the role object / role store)

I think Role might be an important one here, as it's not unlikely that errors are thrown due to a specific permissions set that the admin needs to reproduce

**Reply by u12206050** [reply]
Correct, role is also very good to have.
Datetime is also very good to have I guess and timezone since the user using it might be in different timezone.

**Reply by u12206050** [reply]
I wonder how much data we can actually send especially if it is being injected into href of mailto:

**Reply by rijkvanzanten** [reply]
Good thinking! Seeing that we're rendering this template out client side, maybe instead of "context" with predefined keys we should "just" make `navigator` available? That way you could set things like `{{ navigator.userAgent }}` 🤔 

> I wonder how much data we can actually send especially if it is being injected into href of mailto:

What do you mean? It might get a little finicky in configuration, but you can pre-fill a whole email body in a `mailto:` link with query params, so in theory you should be able to send whatever you want? 🤔

**Reply by u12206050** [reply]
Ok, yeh sure. Think mailto is safest, since the user could be offline actually.

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here: https://github.com/directus/directus/issues/17287
